### PR TITLE
feat: create post

### PR DESCRIPTION
**Pull request by sshanzel**
## Changes
- Create a freeform post.
- Edit a freeform post.
- Created a hook to reuse the notification toggle functionality as it seems to be more frequent now.
- Updated the ImageInput to be "closeable".
- Fixed some typo
- Created a new hook useSquad which returns an `isForbidden` property out of the box and avoids the unnecessary state. (tested and still works as intended).
- Refactored the unnecessary usage of state for firing tracking events.

**Note:** the freeform post modal and card have their own dedicated tasks, so there are rough edges when viewing the post page of the newly created freeform post but it is for the sake of testing.

You might wonder if some parts are missing like the image uploads on the content, and saving on the cache. Those things can be worked on in parallel and have their own tasks too. Once this is merged, those areas should be unblocked to anyone who wants to work with v6.

### Describe what this PR does
- Short and concise, bullet points can help
- Screenshots if applicable can also help

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-{number} #done

### CODE REVIEW

File: packages/shared/src/components/fields/ImageInput.tsx [hunk 1]
```diff
@@ -22,17 +24,21 @@ interface ClassName {
...
-  alwayShowHover?: boolean;
+  enableHover?: boolean;
+  closeable?: boolean;
+  alwaysShowHover?: boolean;
+  children?: ReactNode;
+  fileSizeLimitMB?: number;
 }
 
-const TWO_MEGABYTES = 2 * 1024 * 1024;
+const MEGABYTE = 2 * 1024 * 1024;
```
Comment (idoshamun) [inline packages/shared/src/components/fields/ImageInput.tsx:41]:
```suggestion
const MEGABYTE = 1024 * 1024;
```

**Comment by sshanzel** [comment]
Have added the edit page and should be working as intended now too.

File: packages/shared/src/components/post/SquadPostContent.tsx [hunk 2]
```diff
@@ -23,6 +23,7 @@ import { SourcePermissions, Squad } from '../../graphql/sources';
 import EditIcon from '../icons/Edit';
 
 const ContentMap = {
+  [PostType.Freeform]: WelcomePostContent,
```
Comment (idoshamun) [inline packages/shared/src/components/post/SquadPostContent.tsx:26]: What does it mean? What is `WelcomePostContent`?

File: packages/shared/src/graphql/posts.ts [hunk 3]
```diff
@@ -487,3 +493,43 @@ interface UpdatePinnedProps {
...
+  ) {
+    createFreeformPost(
+      sourceId: $sourceId
+      title: $title
+      content: $content
+      image: $image
+    ) {
+      ...SharedPostInfo
+      trending
+      views
```
Comment (idoshamun) [inline packages/shared/src/graphql/posts.ts:512]: These are not needed right? (trending and views)

File: packages/shared/src/graphql/posts.ts [hunk 4]
```diff
@@ -487,3 +493,43 @@ interface UpdatePinnedProps {
...
+      contentHtml
+      source {
+        ...SourceBaseInfo
+      }
+      description
+      summary
+      toc {
+        text
+        id
+      }
```
Comment (idoshamun) [inline packages/shared/src/graphql/posts.ts:523]: Also these

**Review by idoshamun — CHANGES_REQUESTED** [review]
Please see my comments

File: packages/shared/src/components/post/SquadPostContent.tsx (hunk 2 above)
Comment (sshanzel) [reply packages/shared/src/components/post/SquadPostContent.tsx:26]: This is just a placeholder to test it, but basically, we have a dedicated ticket to introduce a `PostContent` dedicated to FreeformPost.

**Review by idoshamun — APPROVED** [review]

**Pull request merged**